doppelganger/doppelganger.go

```diff
@@ -309,6 +315,23 @@ func (h *handler) resetDoppelgangerStates() {
 	h.logger.Info("All Doppelganger states reset to initial detection epochs")
 }
 
+func (h *handler) recordValidatorStates(ctx context.Context) {
+	h.mu.RLock()
+	defer h.mu.RUnlock()
+
+	var safe, unsafe uint64
+	for _, state := range h.validatorsState {
+		if state.safe() {
+			safe++
+		} else {
+			unsafe++
+		}
+	}
```

Comment:
i suggest this to free the lock sooner for potential signing requests:
```suggestion
	var safe, unsafe uint64
	func() {
		h.mu.RLock()
		defer h.mu.RUnlock()

		for _, state := range h.validatorsState {
			if state.safe() {
				safe++
			} else {
				unsafe++
			}
		}
	}()
```